lib/samples/run_valve_isolation_trace/run_valve_isolation_trace.dart

```diff
@@ -0,0 +1,580 @@
+                      child: Text(
+                        category.name,
+                        style: Theme.of(context).textTheme.bodyMedium,
+                      ),
+                    );
+                  }).toList(),
+                  onChanged: (value) {
+                    setState(() {
+                      _selectedCategory = value;
+                    });
+                  },
+                ),
+              ),
+              // The button to start tracing.
+              ElevatedButton(
+                onPressed: _traceEnabled ? _onTrace : null,
+                child: const Text('Trace'),
+              ),
+              // The button to reset the trace.
+              ElevatedButton(
+                onPressed: _resetEnabled ? _clear : null,
+                child: const Text('Reset'),
+              ),
+            ],
+          ),
+          // The Switch for including isolated features.
+          Row(
+            spacing: 4,
+            children: [
+              Text(
+                'Include isolated features',
+                style: TextStyle(
+                  color: (_selectedCategory == null)
+                      ? Colors.grey
+                      : Theme.of(context).textTheme.bodyMedium?.color,
+                ),
+              ),
+              Switch(
+                value: _isIncludeIsolatedFeatures,
+                onChanged: (v) =>
+                    setState(() => _isIncludeIsolatedFeatures = v),
+              ),
+              if (_isIncludeIsolatedFeatures)
+                const Text('On')
+              else
+                const Text('Off'),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+
+  /// Called when the map view is ready.
+  Future<void> _onMapViewReady() async {
+    setState(() {
+      _loading = true;
+      _statusMessage = 'Loading map...';
+    });
+
+    final map = ArcGISMap.withUri(
+      Uri.parse(
+        'https://sampleserver7.arcgisonline.com/portal/home/item.html?id=f439b4724bb54ac088a2c21eaf70da7b',
+      ),
+    );
+    await map!.load();
+    _mapViewController.arcGISMap = map;
+
+    // Load the utility network.
+    try {
+      await _loadUtilityNetwork(map);
+    } on Exception catch (e) {
+      setState(() {
+        _statusMessage = 'Error loading Utility Network: $e';
+      });
+    }
+    setState(() => _loading = false);
+  }
+
+  /// Load the utility network from the service geodatabase.
+  Future<void> _loadUtilityNetwork(ArcGISMap map) async {
+    setState(() => _statusMessage = 'Loading Utility Network...');
+
+    // Get and load the utility network from the map.
+    _utilityNetwork = map.utilityNetworks.first;
+    await _utilityNetwork.load();
+
+    // Get a trace configuration from a tier.
+    final domainNetwork = _utilityNetwork.definition?.getDomainNetwork(
+      'Pipeline',
+    );
+    final tier = domainNetwork?.getTier('Pipe Distribution System');
+    // A tier must be selected
```

Comment:
This comment doesn't have a period at the end, but I don't think we really need a comment here. Maybe just delete this comment?